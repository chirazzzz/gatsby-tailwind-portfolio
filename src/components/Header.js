import React from 'react'
import { Link } from 'gatsby'

import { navLinks, siteShortTitle } from '../config'

const Header = () => {
  const { menu } = navLinks
  return (
    <header className="flex items-center justify-between px-12 py-6 border-t-4 border-amber-500">
      <Link to="/" aria-label="home">
        <h3 className="text-4xl font-black">
          {siteShortTitle}
          <span className="text-amber-500 font-medium">.</span>
        </h3>
      </Link>
      <nav className="flex items-center">
        {menu.map(({ name, url }, key) => {
          return (
            
            <Link
              className="text-lg font-bold px-3 py-1 rounded-lg hover:bg-amber-200 hover:text-slate-800 focus:text- duration-200 ease-in-out"
              key={key}
              to={url}
            >
              {name}
            </Link>
          )
        })}
      </nav>
    </header>
  )
}

export default Header
