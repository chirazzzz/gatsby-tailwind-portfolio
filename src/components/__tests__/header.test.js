import React from "react"
import renderer from "react-test-renderer"
import { render } from '@testing-library/react'

import Header from "../Header"

describe("Header", () => {
  it("renders correctly in Jest test", () => {
    const tree = renderer
      .create(<Header siteTitle="Default Starter" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly in React Testing Library', () => {
    const container = render(<Header />)
    expect(container).toMatchSnapshot()
  })
})