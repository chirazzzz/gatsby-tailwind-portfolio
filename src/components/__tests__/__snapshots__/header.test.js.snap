// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly in Jest test 1`] = `
<header
  className="flex items-center justify-between px-12 py-6 border-t-4 border-amber-500"
>
  <a
    aria-label="home"
    href="/"
  >
    <h3
      className="text-4xl font-black"
    >
      CM
      <span
        className="text-amber-500 font-medium"
      >
        .
      </span>
    </h3>
  </a>
  <nav
    className="flex items-center"
  >
    <a
      className="text-lg font-bold px-3 py-1 rounded-lg hover:bg-amber-200 hover:text-slate-800 focus:text- duration-200 ease-in-out"
      href="/#about"
    >
      About
    </a>
    <a
      className="text-lg font-bold px-3 py-1 rounded-lg hover:bg-amber-200 hover:text-slate-800 focus:text- duration-200 ease-in-out"
      href="/#projects"
    >
      Projects
    </a>
    <a
      className="text-lg font-bold px-3 py-1 rounded-lg hover:bg-amber-200 hover:text-slate-800 focus:text- duration-200 ease-in-out"
      href="/#contact"
    >
      Contact
    </a>
  </nav>
</header>
`;
